src/Settings.php

```diff
@@ -117,102 +125,202 @@ public function update( $settings ) {
+					'label'             => __( 'Phone number', 'siwk' ),
+					'default'           => 'yes',
+					'disabled'          => true,
+					'class'             => 'siwk',
+					'custom_attributes' => array(
+						'checked' => 'true',
+					),
+				),
+				'siwk_required_scopes_billing_address' => array(
+					'name'              => 'siwk_required_scopes_billing_address',
+					'type'              => 'checkbox',
+					'label'             => __( 'Billing address', 'siwk' ),
+					'description'       => __( 'These scopes are included by default, as necessary for creating a WooCommerce customer account in your shop.  More about available scopes with Sign in with Klarna <a href="https://docs.klarna.com/conversion-boosters/sign-in-with-klarna/integrate-sign-in-with-klarna/web-sdk-integration/#scopes-and-claims">here</a>. Additional scopes can be customized if applicable, more info <a href="https://gist.github.com/mntzrr/4bf23ca394109d40575f2abc05811ddc">here</a>.', 'siwk' ),
+					'disabled'          => true,
+					'class'             => 'siwk',
+					'custom_attributes' => array(
+						'checked' => 'true',
+					),
+				),
+				'siwk_optional_scopes_date_of_birth'   => array(
+					'name'    => 'siwk_optional_scopes_date_of_birth',
+					'title'   => 'Request Additional Customer Data',
+					'type'    => 'checkbox',
+					'label'   => __( 'Date of birth', 'siwk' ),
+					'default' => 'no',
+				),
+				'siwk_optional_scopes_country'         => array(
+					'name'    => 'siwk_optional_scopes_country',
+					'type'    => 'checkbox',
+					'label'   => __( 'Country', 'siwk' ),
+					'default' => 'no',
+					'class'   => 'siwk',
+				),
+				'siwk_optional_scopes_language'        => array(
+					'name'    => 'siwk_optional_scopes_language',
+					'type'    => 'checkbox',
+					'label'   => __( 'Language Preference', 'siwk' ),
+					'default' => 'no',
+					'class'   => 'siwk',
+				),
+				'siwk_previews'                        => array(
+					'type'     => 'kp_section_end',
+					'previews' => array(
+						array(
+							'title' => __( 'Preview', 'siwk' ),
+							'image' => $this->get_preview_image(),
+						),
+					),
+				),
+			)
 		);
+	}
 
-		$settings['siwk_region'] = array(
-			'name'        => 'siwk_region',
-			'title'       => __( 'Region', 'siwk' ),
-			'type'        => 'select',
-			'description' => __( 'The regional endpoint.', 'siwk' ),
-			'default'     => $this->default()['siwk_region'],
-			'options'     => array(
-				'eu' => __( 'EU', 'siwk' ),
-			),
-		);
+	/**
+	 * Retrieve the client ID from the Klarna plugin settings.
+	 *
+	 * @param array $settings The settings to extract from.
+	 * @return string The client ID or empty string.
+	 */
+	private function get_client_id( $settings ) {
+		$country             = strtolower( kp_get_klarna_country() );
+		$available_countries = $settings['available_countries'] ?? array();
+		if ( ! in_array( $country, $available_countries, true ) ) {
+			return '';
+		}
```

Comment:
Is there a reason for the custom logic here over using the [kp_get_client_id](https://github.com/krokedil/klarna-payments-for-woocommerce/blob/master/includes/kp-functions.php) function?

Reply:
No, I just wasn't aware of that function. I'll use it :)